src/nspawn/nspawn-oci.c

```diff
@@ -444,14 +444,30 @@ static int oci_process(const char *name, JsonVariant *v, JsonDispatchFlags flags
 }
 
 static int oci_root(const char *name, JsonVariant *v, JsonDispatchFlags flags, void *userdata) {
+        Settings *s = userdata;
+        int r;
 
         static const JsonDispatch table[] = {
                 { "path",     JSON_VARIANT_STRING,  json_dispatch_string,  offsetof(Settings, root)      },
                 { "readonly", JSON_VARIANT_BOOLEAN, json_dispatch_boolean, offsetof(Settings, read_only) },
                 {}
         };
 
-        return json_dispatch(v, table, oci_unexpected, flags, userdata);
+        r = json_dispatch(v, table, oci_unexpected, flags, userdata);
```

Comment:
hmm, let's pass `s` instead of `userdata` here as last argument. of course, that's 100% the same, but I think when we have a typed alias variable of userdata already, we should only use that and forget entirely about userdata, just as a matter of style.